clinica/iotools/utils/pipeline_handling.py

```diff
@@ -4,388 +4,274 @@
-                        )
-                        if os.path.exists(atlas_path):
-                            n_regions = summary_df[
-                                (summary_df.group_id == group)
-                                & (summary_df.atlas_id == atlas)
-                            ]["regions_number"].values[0]
-                            atlas_df = pd.read_csv(atlas_path, sep="\t")
-                            label_list = [
-                                group + "_" + atlas + "_ROI-" + str(x)
-                                for x in range(n_regions)
-                            ]
-                            ses_df.loc[:, label_list] = atlas_df[
-                                "mean_scalar"
-                            ].to_numpy()
 
-        pipeline_df.iloc[i] = ses_df.iloc[0]
+        pipeline_df = pipeline_df.append(ses_df)
 
+    summary_df = generate_summary(pipeline_df, "t1-freesurfer", ignore_groups=True)
     final_df = pd.concat([df, pipeline_df], axis=1)
 
     return final_df, summary_df
 
 
-def t1_volume_pipeline(caps_dir, df, **kwargs):
+def t1_volume_pipeline(caps_dir, df, group_selection=None, volume_atlas_selection=None, **kwargs):
     """Merge data of the t1-volume pipeline to the merged file containing the BIDS information.
 
     Args:
         caps_dir: the path to the CAPS directory
         df: the DataFrame containing the BIDS information
-        atlas_selection: allows to choose the atlas to merge. If None all atlases are selected.
+        group_selection: allows to choose the DARTEL groups to merge. If None all groups are selected.
+        volume_atlas_selection: allows to choose the atlas to merge. If None all atlases are selected.
 
     Returns:
         final_df: a DataFrame containing the information of the bids and the pipeline
     """
-    from ...pipelines.t1_volume.t1_volume_cli import T1VolumeCLI
+    t1_spm_path = path.join("t1", "spm", "dartel")
 
-    instance = T1VolumeCLI()
-    pipeline_name = instance.name
-    del instance
+    return volume_pipeline(caps_dir, df, t1_spm_path,
+                           group_selection=group_selection,
+                           atlas_selection=volume_atlas_selection,
+                           pvc_restriction=None,
+                           pipeline_name="t1-volume")
 
-    # Recovering optional arguments
-    if "atlas_selection" in kwargs.keys():
-        atlas_selection = kwargs["atlas_selection"]
-    else:
-        atlas_selection = None
 
-    if "group_selection" in kwargs.keys():
-        group_selection = kwargs["group_selection"]
-    else:
-        group_selection = None
+def volume_pipeline(caps_dir, df, pipeline_path, pipeline_name,
+                    group_selection=None, atlas_selection=None,
+                    pvc_restriction=None, tracers_selection=None):
+    """Merge data of the t1-volume pipeline to the merged file containing the BIDS information.
```

Comment:
```suggestion
    """Merge data of the t1-volume and pet-volume pipelines to the merged file containing the BIDS information.
```

Reply:
Done in https://github.com/aramis-lab/clinica/pull/166/commits/4522a7d336c36b2b149249362a839d4033ff8517